src/main/java/com/github/pires/obd/commands/control/TroubleCodesCommand.java

```diff
@@ -54,14 +54,17 @@ protected void performCalculations() {
         final String result = getResult();
         String workingData;
         int startIndex = 0;//Header size.
-        if (result.length() % 4 == 0) {//CAN(ISO-15765) protocol one frame.
-            workingData = getResult();//43yy{codes}
+
+        String canOneFrame = result.replaceAll("[\r\n]", "");
+        int canOneFrameLength = canOneFrame.length();
+        if (canOneFrameLength <= 16 && canOneFrameLength % 4 == 0) {//CAN(ISO-15765) protocol one frame.
+            workingData = canOneFrame;//43yy{codes}
             startIndex = 4;//Header is 43yy, yy showing the number of data items.
         } else if (result.contains(":")) {//CAN(ISO-15765) protocol two and more frames.
-            workingData = getResult().replaceAll("[\r\n].:", "");//xxx43yy{codes}
+            workingData = result.replaceAll("[\r\n].:", "");//xxx43yy{codes}
```

Comment:
Isn't this done already with `String canOneFrame = result.replaceAll("[\r\n]", "");`?